src/framework/emqttb_worker.erl

```diff
@@ -444,6 +446,13 @@ connect_fun()->
             connect
     end.
 
+-spec broker_host(integer()) -> {string(), inet:port_number()}.
```

Comment:
On the one hand, it'll be a bit more non-deterministic, since if the client fails to connect to the head broker, it'll connect to the next one, and lose the desired forwarding property.  Also, it might make the connection number per broker a bit more uneven.

On the other hand, if a single client fails to connect, dies, and a new one is spawned by the group, then it'll have a totally different ID number and might connect to any broker.  

I still don't have a good idea to solve this.  Maybe if we kept connection counter for each host, then try to balance that.  But that's also very complicated.  :thinking:

Reply:
Other scenarios may require connecting to the hosts in a different way, e.g. if we test node restart under traffic, smth. like round-robin is needed. 
I don't see a better option other than making host selection function configurable (per group). E.g. offer several strategies:
- Round-robin
- Random
- Deterministic (your implementation)